File: de.lmu.ifi.dbs.knowing.core.swt/src/de/lmu/ifi/dbs/knowing/core/swt/view/PresenterView.java
/**
 * 
 */
package de.lmu.ifi.dbs.knowing.core.swt.view;

import org.eclipse.swt.SWT;
import org.eclipse.swt.layout.FillLayout;
import org.eclipse.swt.widgets.Composite;
import org.eclipse.swt.widgets.TabFolder;
import org.eclipse.swt.widgets.TabItem;
import org.eclipse.ui.part.ViewPart;

/**
 * @author Nepomuk Seiler
 * @version 0.1
 * @since 21.03.2011
 */
public class PresenterView extends ViewPart {

	public static final String ID = "de.lmu.ifi.dbs.knowing.core.swt.presenterView";
	
	private TabFolder tabFolder;

	@Override
	public void createPartControl(Composite parent) {
		tabFolder = new TabFolder(parent, SWT.BOTTOM);
	}
	
	/**
	 * <p>Creates a tab with the given name and returns <br>
	 * the composite which is controlled by the tabitem.</p>
	 * 
	 * @param name - name of the tab
	 * @return tab content composite
	 */
	public Composite createTab(String name) {
		TabItem tabItem = new TabItem(tabFolder, SWT.NONE);
		tabItem.setText(name);
		Composite composite = new Composite(tabFolder, SWT.NONE);
		composite.setLayout(new FillLayout(SWT.HORIZONTAL));
		tabItem.setControl(composite);
		return composite;
	}
	
	public void clearTabs() {
		TabItem[] items = tabFolder.getItems();
		for (TabItem tabItem : items) 
			tabItem.dispose();
	}

	@Override
	public void setFocus() {
		tabFolder.setFocus();
	}

}

File: de.lmu.ifi.dbs.knowing.core.swt/src/de/lmu/ifi/dbs/knowing/core/swt/view/PresenterView.scala
package de.lmu.ifi.dbs.knowing.core.swt.view

import akka.actor.{ ActorRef, Actor, TypedActor }
import org.eclipse.swt.SWT
import org.eclipse.swt.widgets.Composite
import org.eclipse.swt.graphics.Image
import org.eclipse.swt.custom.{ CTabFolder, CTabItem }
import org.eclipse.ui.part.ViewPart
import de.lmu.ifi.dbs.knowing.core.swt.dialog.ProgressDialog
import de.lmu.ifi.dbs.knowing.core.swt.factory.UIFactories.{ newTabUIFactoryInstance, newServiceProperties }
import de.lmu.ifi.dbs.knowing.core.factory.UIFactory
import de.lmu.ifi.dbs.knowing.core.events._
import de.lmu.ifi.dbs.knowing.core.model._
import de.lmu.ifi.dbs.knowing.core.swt.internal.Activator
import org.osgi.framework.ServiceRegistration
import com.typesafe.config.ConfigFactory
import akka.actor.ActorSystem

/**
 * @author Nepomuk Seiler
 * @version 0.3
 * @since 22.04.2011
 *
 */
class PresenterView extends ViewPart {

	var uifactory: UIFactory[Composite] = _

	private var uiFactoryReg: ServiceRegistration[UIFactory[Composite]] = _

	/**
	 * Creates UIFactory and register it as a service
	 */
	def createPartControl(parent: Composite) = {
		val asm = Activator.actorSystemManager
		val system = asm.getSystem("swt-local").
			getOrElse(asm.create("swt-local", ConfigFactory.defaultReference(classOf[ActorSystem].getClassLoader)))

		uifactory = newTabUIFactoryInstance(system, parent, PresenterView.ID)

		//Register UIFactory as a service
		val ctx = Activator.getDefault.getBundle.getBundleContext
		uiFactoryReg = ctx.registerService(classOf[UIFactory[Composite]], uifactory, newServiceProperties)
	}

	def setFocus() = {}

	override def dispose() {
		uiFactoryReg.unregister
		super.dispose()
	}

}

object PresenterView { val ID = "de.lmu.ifi.dbs.knowing.core.swt.presenterView" }

